packages/edit-site/src/components/sidebar-navigation-screen-navigation-menu/index.js

```diff
@@ -229,7 +229,7 @@ export default function SidebarNavigationScreenNavigationMenu() {
 				'Navigation menus are a curated collection of blocks that allow visitors to get around your site.'
 			) }
 		>
-			<NavigationMenuEditor navigationMenu={ navigationMenu } />
+			<NavigationMenuEditor navigationMenuId={ navigationMenu?.id } />
```

Comment:
Always better to pass simpler props as they are less likely to trigger re-renders.